src/helpers/serializer.ts

```diff
@@ -0,0 +1,91 @@
+import * as firebase from 'firebase/app'
+import {
+  CollectionQueryType,
+  WhereArray,
+  WhereItem,
+  WhereType,
+} from '../types/Query'
+
+export class Serializer {
+  // Helper for where clauses
+  public static multipleConditions<Doc extends object = {}>(
+    w: WhereType<Doc>
+  ): w is WhereArray<Doc> {
+    return !!(w as WhereArray) && Array.isArray(w[0])
+  }
+
+  // Serializer function for where condition
+  private static serializeWhere<Doc extends object = {}>(
+    where: WhereType<Doc>
+  ): WhereType<Doc> {
+    if (this.multipleConditions(where)) {
+      return where.map(w => this.serializeWhere(w)) as WhereArray<Doc>
+    }
+    // Date: Inject serializer options if not specified
+    if (where[2] instanceof Date && !where[3]) {
+      return [...where.slice(0, 3), { type: 'date' }] as WhereItem<Doc>
+    }
+
+    if (where[2] instanceof firebase.firestore.DocumentReference && !where[3]) {
```

Comment:
@nandorojo I am pretty sure this breaks compatibility with `8.x` because when I tried using this in my `8.x` firebase project, I got `cannot read property "DocumentReference" of undefined`.
I think it is best to create a new major version and convert the project to support firebase `8.x` instead, so people do not have to use their own `Fuego` like I have to do today.

Reply:
Temporarily I reverted my project to `7.x`, but the downside is firestore does not have a `useEmulator` function there. It's possible to set settings manually which I have done in my project right now and it is working, but I do want to use `8.x` in my project. What do you think?